test/index.js

```diff
@@ -199,56 +199,100 @@ describe('Intercom', function(){
+        .end(function(err, res){
+          if (err) return err;
+          // save jobId for later tests
+          jobId = res[0].res.body.id;
+          done();
+        });
+    });
 
-      var input = test.fixture('group-basic').input;
-      input.traits.created_at = time(new Date(input.traits.created_at));
+    it('should add to job if already exists', function(done){
+      var json = test.fixture('group-job-existing');
+      json.input.userId = userId;
+      json.output.items[0].data.user_id = userId;
+      json.output.job = { id: jobId };
 
-      var name = input.traits.name;
-      delete input.traits.name;
+      test
+        .set(settings)
+        .group(json.input)
+        .sends(json.output)
+        .expects(200)
+        .end(done);
+    });
 
-      var payload = {};
-      payload.user_id = input.userId;
-      payload.last_request_at = time(input.timestamp);
-      payload.companies = [{
-        company_id: input.groupId,
-        custom_attributes: input.traits,
-        name: name,
-        remote_created_at: input.traits.created_at
-      }];
+    it('should work with .created_at', function(done){
+      var json = test.fixture('group-job-created_at');
+      json.input.userId = userId;
+      json.output.items[0].data.user_id = userId;
+      json.output.job = { id: jobId };
 
-      payload.custom_attributes = {
-        id: input.userId
-      };
+      test
+        .set(settings)
+        .group(json.input)
+        .sends(json.output)
+        .expects(200)
+        .end(done);
+    });
+
+    it('should work with .createdAt', function(done){
+      var json = test.fixture('group-job-createdAt');
+      json.input.userId = userId;
+      json.output.items[0].data.user_id = userId;
+      json.output.job = { id: jobId };
 
       test
-        .request(1)
-        .sends(payload)
-        .expects(200);
+        .set(settings)
+        .group(json.input)
+        .sends(json.output)
+        .expects(200)
+        .end(done);
+    });
 
-      test.end(done);
-    })
+    it('should just create a new job if adding to a job fails', function(done){
+      var json = test.fixture('group-job-existing');
+      json.input.userId = userId;
+      json.output.items[0].data.user_id = userId;
 
-    it('should work with .created_at', function(done){
-      var traits = { created_at: 'Jan 1, 2000 3:32:33 PM', name: 'old company' };
-      var group = helpers.group({ traits: traits, groupId: 'a5322d6' });
-      delete group.obj.traits.created;
-      intercom.group(group, done);
-    })
-
-    it('should work with .created', function(done){
-      var traits = { created: 'Jan 1, 2014 3:32:33 PM', name: 'new company' };
-      var group = helpers.group({ traits: traits, groupId: 'e186e5de' });
-      intercom.group(group, done);
-    })
+      // Modify valid jobId stored in redis to be invalid
+      var jobKey = [settings.appId, 'jobs', userId].join(':');
+      intercom.redis().set(jobKey, 'garbage_id');
```

Comment:
This isn't guaranteed to finish until the callback is called. You should move the rest of this test inside a callback that you pass to `redis().set` to avoid sporadic failures. (Pretty sure I missed this in the track PR, so that needs to be changed as well)